client.go

```diff
@@ -564,12 +570,18 @@ func (cl *Client) dialFirst(addr string, t *Torrent) (conn net.Conn, utp bool) {
 	return
 }
 
-func (cl *Client) noLongerHalfOpen(t *Torrent, addr string) {
+func (cl *Client) noLongerHalfOpen(t *Torrent, addr string) bool {
```

Comment:
I think I'd rather just leave the halfOpen field even when the torrent is closed. No need to return anything. Just check for torrent closed when it matters?